### PR TITLE
Support html-webpack-plugin v4

### LINKED ISSUE #128: Support for html-webpack-plugin 4?
It's still not out quite yet, but I'm using html-webpack-plugin 4.0.0-alpha.2 for performance reasons. I can't seem to get this plugin working with it, so I'm guessing it's not supported. The integration API has changed considerably:

https://github.com/jantimon/html-webpack-plugin/pull/953

If you could support both versions, that would be much appreciated. 

Cheers!

### PR DESCRIPTION
**Pull request by brunocodutra**
Closes #128

### CODE REVIEW

File: src/compat.js [hunk 1]
```diff
@@ -14,6 +14,14 @@ module.exports.tap = (tappable, hook, name, plugin) => (
   : tappable.plugin(hook, plugin)
 );
 
+module.exports.tapHtml = (tappable, name, plugin) => {
+  const HtmlWebpackPlugin = require('html-webpack-plugin');
```
Comment (keepitterron) [inline src/compat.js:18]: It's using `"html-webpack-plugin": "^3.2.0"` from `package.json` here.
`HtmlWebpackPlugin.getHooks` will always be undefined.

Or at least that's what I'm noticing testing it locally.
Makes sense it's undefined tho, the version with getHooks is `html-webpack-plugin@next`.

Comment (brunocodutra) [reply src/compat.js:18]: Unless I misunderstood, `html-webpack-plugin` ist a dev-dependency and should not be installed if you consume `webapp-webpack-plugin` as a dependency, meaning it should pick whichever version you have in your package.json.

Comment (keepitterron) [reply src/compat.js:18]: I think you're right, my understanding of npm dependencies is lacking at best.

**Review by keepitterron — APPROVED** [review]
Works well for me!

**Comment by bj00rn** [comment]
@brunocodutra can we have this merged?

**Comment by brunocodutra** [comment]
@bj00rn yes! The only reason I didn't merge it yet though is the fact it lacks unit tests and it's getting increasingly hard to set them up such that it works for legacy versions of Webpack and HtmlWebpackPlugin, so I've been fiddling with the idea of releasing a major version.

**Pull request merged**